runtime/lua/vim/treesitter/dev.lua

```diff
@@ -164,8 +164,13 @@ function TSTreeView:draw(bufnr)
   for _, item in self:iter() do
     local range_str = get_range_str(item.lnum, item.col, item.end_lnum, item.end_col)
     local lang_str = self.opts.lang and string.format(' %s', item.lang) or ''
+    local text = item.text
+    if not item.named then
+      text = '"' .. text:sub(2, #text - 1):gsub('"', '\\"') .. '"'
+    end
```

Comment:
- ternary is less verbose and more readable
- we need to stop using `..` to concat strings, format() tends to be more readable.

```suggestion
    local text = item.named and item.text or ('"%s"'):format(text:sub(2, #text - 1):gsub('"', '\\"'))
```

Reply:
Personally I think the suggestion has way too much going on in one line.

Reply:
the point of the suggestion is obvious, the bugs are easy to fix

Reply:
Which of these two do you find more readable?

```lua
    local text = item.named and item.text or
      string.format('"%s"', item.text:sub(2, #item.text - 1):gsub('"', '\\"'))
```

```lua
    local text = item.text
    if not item.named then
      text = string.format('"%s"', text:sub(2, #text - 1):gsub('"', '\\"'))
    end
```

Reply:
Neither. Thus the one that only does one assignment is preferable for the reason I mentioned.

Reply:
> do I really have run every suggestion through a compiler?

Sorry I wasn't clear enough. I was trying to imply that the corrected suggestion would have been even more complicated.

> text = '"' .. text:sub(2, #text - 1):gsub('"', '\\"') .. '"' doesn't become clearer just by more verbose control-flow...

I disagree, but of course this is subjective. Overly complex expressions are just hard to read. Breaking it into an if-chain just makes it much easier to differentiate conditions with values.

> reassigning a variable is like "aliasing", it makes the content of a variable harder to follow. when it can easily be avoided w/ a ternary, it's usually a win.

I agree, atm the moment for complex initializers I do something like.

```lua
local foo --- @type TYPE
if cond then
   ...
   foo = 
else
   ...
   foo = 
end
```

And typical I would then just factor this into a function.

I think the problem is that it isn't obvious enough what `string.format('"%s"', text:sub(2, #text - 1):gsub('"', '\\"'))` is trying to achieve without reading every term. For that reason I would put this in a function like `escape_quotes()` or something which would give us:

```lua
 local text = item.named and item.text or escape_quotes(item.text)
```